plugins/deps/src/main/java/dependencies.kt

```diff
@@ -33,15 +34,26 @@ val DepType.files: List<FileSpec>
 
 val Provider<out Dependency>.dep: NameSpec
     get() {
-        val pluginConf = FormaSettingsStore.pluginFor(this)
+        val depName = get().run { "$group:$name:$version" }
+        val pluginConf = FormaSettingsStore.pluginFor(depName)
+        pluginConf?.let { println("PLUGIN CONFIGURATION: $it") }
+        println(this.get())
         return with(get()) {
             NameSpec(
                 "$group:$name:$version",
-                pluginConf?.configuration?.let(::Custom) ?: Implementation
+                pluginConf?.configuration?.let(::CustomConfiguration) ?: Implementation
             )
         }
     }
 
+fun Provider<out Dependency>.dep(configuration: CustomConfiguration): NameSpec {
+    return with(get()) { NameSpec("$group:$name:$version", configuration) }
+}
```

Comment:
does we really need to do it by ourself? maybe We can do it without calling `get()` and pass it to gradle as `Provider` ?

Reply:
yep, will do that once once everything else works